### ISSUE TITLE
[Feature] url support for unarchive module

### ISSUE THREAD

**Issue by morgajel** [ansible/ansible-modules-core#1443]
##### Issue Type:

Feature Idea
##### Ansible Version:

ansible 1.9.1
##### Environment:

Mint 17 host, Centos7 target
##### Summary:

It would be nice to specify a url as the source in an unarchive line:
    - unarchive: src=http://edcint.co.nz/checkwmiplus/sites/default/files/check_wmi_plus.v1.59.tar.gz  dest=/root creates=/opt/check_wmi_plus

Note that the yum module currently does this (although that's yum's behavior, not ansible). It would be nice to have some consistency in possible sources.

Of course the current workaround is to copy the file from the url to the remote system first, then unarchive it, but this seems less intuitive.

Just throwing it out there; I can see pros and cons for both sides.

**Comment by Jmainguy** [comment]
Coded this into the yum module for EL5 and below, by having python download the file, should be able to port that code to this module.

**Comment by Jmainguy** [comment]
@morgajel give #1448 a shot, should do what you need. Make sure copy=no if using a url

**Comment by akaihola** [comment]
This doesn't seem to work. I always get

```
fatal: [myhost]: FAILED! => {"msg": "ERROR! file or module does not exist: /home/akaihola/playbooks/https:/myserver.com/file.zip", "failed": true}
```

I'm at commit 6288d44581e1d6eb5505d52ac692a9ef8f6b4918.

**Comment by Jmainguy** [comment]
@akaihola did you try with copy=no, yes is the default, and if yes it doesnt try to download it, on the off chance that someone actuall has :// in their filename and wants to copy it over

**Comment by akaihola** [comment]
@Jmainguy yes, I have `copy=no`. The strange thing is, it works if I run it with `test-module`. As soon as I run `ansible-playbook`, it fails. I've triple-checked that I'm actually running the same code in both cases.

**Comment by Jmainguy** [comment]
@akaihola  does your playbook have :// in the url, or just :/ looks like I am looking for elif '://' in src:

**Comment by akaihola** [comment]
@Jmainguy my URL starts with `https://`

**Comment by akaihola** [comment]
@Jmainguy these are my actual tasks:

```
- name: create pip installation directory
  file: path=/tmp/pip-7.0.3
        state=directory

- name: fetch pip
  unarchive: src=https://pypi.python.org/packages/source/p/pip/pip-7.0.3.tar.gz dest=/tmp/pip-7.0.3 copy=no
```

**Comment by akaihola** [comment]
And here is the output using the `-vvvv` verbose flag:

```
TASK [install-pip : install-pip : fetch pip] ****************
<myhost.domain.com> ESTABLISH SSH CONNECTION FOR USER: root
<myhost.domain.com> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/tmp/ansible-ssh-%h-%p-%r" -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 myhost.domain.com mkdir -p $HOME/.ansible/tmp/ansible-tmp-1435609950.05-95151960477187 && echo $HOME/.ansible/tmp/ansible-tmp-1435609950.05-95151960477187
<myhost.domain.com> ESTABLISH SSH CONNECTION FOR USER: root
<myhost.domain.com> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/tmp/ansible-ssh-%h-%p-%r" -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 myhost.domain.com rc=flag; [ -r /tmp/pip-7.0.3/ ] || rc=2; [ -f /tmp/pip-7.0.3/ ] || rc=1; [ -d /tmp/pip-7.0.3/ ] && rc=3; python -V 2>/dev/null || rc=4; [ x"$rc" != "xflag" ] && echo "${rc}  "/tmp/pip-7.0.3/ && exit 0; (python -c 'import hashlib; BLOCKSIZE = 65536; hasher = hashlib.sha1();
afile = open("'/tmp/pip-7.0.3/'", "rb")
buf = afile.read(BLOCKSIZE)
while len(buf) > 0:
    hasher.update(buf)
    buf = afile.read(BLOCKSIZE)
afile.close()
print(hasher.hexdigest())' 2>/dev/null) || (python -c 'import sha; BLOCKSIZE = 65536; hasher = sha.sha();
afile = open("'/tmp/pip-7.0.3/'", "rb")
buf = afile.read(BLOCKSIZE)
while len(buf) > 0:
    hasher.update(buf)
    buf = afile.read(BLOCKSIZE)
afile.close()
print(hasher.hexdigest())' 2>/dev/null) || (echo '0  '/tmp/pip-7.0.3/)
<myhost.domain.com> ESTABLISH SSH CONNECTION FOR USER: root
<myhost.domain.com> PUT /home/akaihola/playbooks/https:/pypi.python.org/packages/source/p/pip/pip-7.0.3.tar.gz TO /root/.ansible/tmp/ansible-tmp-1435609950.05-95151960477187/source
fatal: [myhost.domain.com]: FAILED! => {"msg": "ERROR! file or module does not exist: /home/akaihola/playbooks/https:/pypi.python.org/packages/source/p/pip/pip-7.0.3.tar.gz", "failed": true}
```

I don't understand the second-last "PUT" line. Where does "/source" come from? Why isn't it copying the script with the name `unarchive` instead?

I also tried quoting the URL, but it didn't help at all.

**Comment by Jmainguy** [comment]
Just installed from devel and I am having the same issue as well, but it works with the copy of the file I had before submitting my PR. Sooooooooooooo I blame @abadger. Looking into it now, ty for reporting this

**Comment by akaihola** [comment]
Ok, I found the reason. After removing `lib/ansible/plugins/action/unarchive.py` and `lib/ansible/plugins/action/unarchive.pyc`, everything works correctly.

**Comment by Jmainguy** [comment]
Can confirm, that fixed it for me as well

**Comment by bcoca** [comment]
closing this via merged #1448
